ts/screens/main/MessagesScreen.tsx

```diff
@@ -1,24 +1,87 @@
-import { Container } from "native-base";
+import { Container, Content, Text } from "native-base";
 import * as React from "react";
-import { NavigationScreenProp, NavigationState } from "react-navigation";
+import { ActivityIndicator } from "react-native";
+import {
+  NavigationEventSubscription,
+  NavigationScreenProp,
+  NavigationState
+} from "react-navigation";
+import { connect } from "react-redux";
+
+import { MessageWithContent } from "../../../definitions/backend/MessageWithContent";
+import { ReduxProps } from "../../actions/types";
+import { GlobalState } from "../../reducers/types";
+import { ServicesListObject } from "../../sagas/messages";
+import { loadMessages } from "../../store/actions/messages";
+import { orderedMessagesSelector } from "../../store/reducers/entities/messages";
+import { createLoadingSelector } from "../../store/reducers/loading";
+
+type ReduxMappedProps = {
+  isLoadingMessages: boolean;
+  messages: ReadonlyArray<MessageWithContent>;
+  servicesById: ServicesListObject;
+};
 
 export type OwnProps = {
   navigation: NavigationScreenProp<NavigationState>;
 };
 
-export type Props = OwnProps;
+export type Props = ReduxMappedProps & ReduxProps & OwnProps;
 
 /**
  * This screen show the messages to the authenticated user.
  */
 class MessagesScreen extends React.Component<Props, never> {
+  private didFocusSubscription:
+    | NavigationEventSubscription
+    | undefined = undefined;
+
   constructor(props: Props) {
     super(props);
   }
 
+  public componentDidMount() {
+    // tslint:disable-next-line
+    this.didFocusSubscription = this.props.navigation.addListener(
+      "didFocus",
```

Comment:
here we probably need a less aggressive approach, like pull to refresh - can you please add a story about that?